src/libraries/System.Data.Odbc/src/Common/System/Data/Common/DBConnectionString.cs

```diff
@@ -308,12 +308,7 @@ private static void ValidateCombinedSet(DBConnectionString? componentSet, DBConn
                         // Combined - Component == null
                         Debug.Assert(!combinedSet._restrictionValues.Except(componentSet._restrictionValues).Any(), "Combined set allows values not allowed by component set");
                     }
-                    else if (combinedSet._behavior == KeyRestrictionBehavior.PreventUsage)
-                    {
-                        // Component==Allow, Combined==PreventUsage
-                        // Preventions override allows, so there is nothing to check here
-                    }
-                    else
+                    else if (combinedSet._behavior != KeyRestrictionBehavior.PreventUsage)
```

Comment:
Different types of changes shouldn't be mixed in one pull request, especially for different areas.

And also, the comment for no-op is valuable and should be preserved in some way.

Reply:
We believe that the empty condition body is misleading and can be replaced with a clearer option.

Reply:
We believe that the empty condition body is misleading and can be replaced with a clearer option.

Reply:
It should not be misleading with comment describing why no action should be taken.